CMakeLists.txt

```diff
@@ -1,53 +1,47 @@
-
-option(VMA_STATIC_VULKAN_FUNCTIONS "Link statically with Vulkan API" ON)
-option(VMA_DYNAMIC_VULKAN_FUNCTIONS "Fetch pointers to Vulkan functions internally (no static linking)" OFF)
-option(VMA_DEBUG_ALWAYS_DEDICATED_MEMORY "Every allocation will have its own memory block" OFF)
-option(VMA_DEBUG_INITIALIZE_ALLOCATIONS "Automatically fill new allocations and destroyed allocations with some bit pattern" OFF)
-option(VMA_DEBUG_GLOBAL_MUTEX "Enable single mutex protecting all entry calls to the library" OFF)
-option(VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT "Never exceed VkPhysicalDeviceLimits::maxMemoryAllocationCount and return error" OFF)
-
-message(STATUS "VMA_STATIC_VULKAN_FUNCTIONS = ${VMA_STATIC_VULKAN_FUNCTIONS}")
-message(STATUS "VMA_DYNAMIC_VULKAN_FUNCTIONS = ${VMA_DYNAMIC_VULKAN_FUNCTIONS}")
-message(STATUS "VMA_DEBUG_ALWAYS_DEDICATED_MEMORY = ${VMA_DEBUG_ALWAYS_DEDICATED_MEMORY}")
-message(STATUS "VMA_DEBUG_INITIALIZE_ALLOCATIONS = ${VMA_DEBUG_INITIALIZE_ALLOCATIONS}")
-message(STATUS "VMA_DEBUG_GLOBAL_MUTEX = ${VMA_DEBUG_GLOBAL_MUTEX}")
-message(STATUS "VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT = ${VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT}")
-
-if(VMA_STATIC_VULKAN_FUNCTIONS OR VMA_BUILD_SAMPLE)
-    find_package(Vulkan REQUIRED)
-else()
-    find_package(VulkanHeaders REQUIRED)
-endif()
+cmake_minimum_required(VERSION 3.23...3.26)
 
-if(VMA_BUILD_SAMPLE)
-    set(VMA_BUILD_SAMPLE_SHADERS ON)
-endif()
+project(VMA LANGUAGES CXX)
+
+add_library(VulkanMemoryAllocator INTERFACE)
+add_library(Vulkan::MemoryAllocator ALIAS VulkanMemoryAllocator)
+
+target_sources(VulkanMemoryAllocator PUBLIC
+    FILE_SET vk_mem_alloc
+    BASE_DIRS include
+    TYPE "HEADERS"
+    FILES include/vk_mem_alloc.h
+)
 
-option(BUILD_DOCUMENTATION "Create and install the HTML based API documentation (requires Doxygen)" OFF)
-
-if(BUILD_DOCUMENTATION)
-    find_package(Doxygen REQUIRED)
-    # set input and output files
-    set(DOXYGEN_IN ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile)
-    set(DOXYGEN_OUT ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
-    # request to configure the file
-    configure_file(${DOXYGEN_IN} ${DOXYGEN_OUT} @ONLY)
-    # note the option ALL which allows to build the docs together with the application
-    add_custom_target( doc_doxygen ALL
-        COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_OUT}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-        COMMENT "Generating API documentation with Doxygen"
-        VERBATIM )
+option(VMA_INSTALL "Install Vulkan Memory Allocator" ${PROJECT_IS_TOP_LEVEL})
+if (VMA_INSTALL)
+    include(GNUInstallDirs)
+
+    set_target_properties(VulkanMemoryAllocator PROPERTIES EXPORT_NAME "MemoryAllocator")
+
+    install(TARGETS VulkanMemoryAllocator EXPORT VulkanMemoryAllocator FILE_SET vk_mem_alloc)
+    install(EXPORT VulkanMemoryAllocator NAMESPACE "Vulkan::" DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/VulkanMemoryAllocator")
```

Comment:
I tested that the following find_package support worked without issue. I can add testing to CI if requested.

```
find_package(VulkanMemoryAllocator CONFIG)
target_link_libraries(foobar PRIVATE Vulkan::MemoryAllocator)
```